tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MongoDB.Driver.Tests
+{
+    public class OcspIntegrationTests
+    {
+        private static readonly string _ocspOutcomeEnvironmentVariableName = "OCSP_TLS_SHOULD_SUCCEED";
+        private readonly ITestOutputHelper _output;
+
+        public OcspIntegrationTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        /*
+         * This test must be run as its own task in EG.
+         * When testing locally, this test should not run alongside other integration tests.
+         * Clearing local OCSP caches between test runs is also required when testing locally.
+         * Clearing the OCSP caches may involve clearing out the caches in ~\.dotnet folder
+         * and $DOTNET_CLI_HOME/.dotnet, as well as running `dotnet clean` prior to running the tests.
+         * See spec for OCSP OS-level cache manipulation commands.
+         *   https://github.com/mongodb/specifications/blob/master/source/ocsp-support/ocsp-support.rst#os-level-ocsp-cache-manipulation
+         * When testing on Windows, the certificate should be added to the trust store prior to each run in order to
+         * reduce the chances of Windows pruning the certificate from the trust store prior to the test running.
+         */
+        [SkippableFact]
+        public void MongoClientShouldRespectCertificateStatusAndTlsInsecure()
+        {
+            /* We cannot call RequireServer.Check() because this would result in a connection being made to the mongod
```

Comment:
If I'm not mistaken. this should work:

    RequireEnvironment.Check().EnvironmentVariable("OCSP_TLS_SHOULD_SUCCEED");

Reply:
Done

Reply:
I think the above comment won't really necessary, we can add a lot of similar comments almost to each test, but we don't do it usually. But I will rely on your opinion here

Reply:
Hm... I think it's worth leaving the comment in since this is a rare test that cannot touch the server AT ALL until the ping command is executed, otherwise the test fails too early. Resolving!